### ISSUE TITLE
haddock build fails

### ISSUE THREAD

**Issue by damiencourousse** [haskell-suite/haskell-src-exts#418]
Using stackage's lts-12.4, which includes version 1.20.3.  
Building Haddock documentation locally with stack, I get this:

```
stack haddock --open haskell-src-exts
haskell-src-exts-1.20.3: configure
haskell-src-exts-1.20.3: build
haskell-src-exts-1.20.3: haddock

--  While building package haskell-src-exts-1.20.3 using:
      /home/damien/.stack/setup-exe-cache/x86_64-linux-tinfo6/Cabal-simple_mPHDZzAJ_2.2.0.1_ghc-8.4.3 --builddir=.stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1 haddock --html --hoogle --html-location=../$pkg-$version/ --haddock-option=--hyperlinked-source --haddock-option=--quickjump
    Process exited with code: ExitFailure (-11)
    Logs have been written to: /home/damien/src/lbc/.stack-work/logs/haskell-src-exts-1.20.3.log

    Configuring haskell-src-exts-1.20.3...
    Preprocessing library for haskell-src-exts-1.20.3..
    Building library for haskell-src-exts-1.20.3..
    [ 1 of 18] Compiling Language.Haskell.Exts.Extension ( src/Language/Haskell/Exts/Extension.hs, .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/build/Language/Haskell/Exts/Extension.o )
    [ 2 of 18] Compiling Language.Haskell.Exts.ExtScheme ( src/Language/Haskell/Exts/ExtScheme.hs, .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/build/Language/Haskell/Exts/ExtScheme.o )
    [ 3 of 18] Compiling Language.Haskell.Exts.SrcLoc ( src/Language/Haskell/Exts/SrcLoc.hs, .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/build/Language/Haskell/Exts/SrcLoc.o )
    [ 4 of 18] Compiling Language.Haskell.Exts.Syntax ( src/Language/Haskell/Exts/Syntax.hs, .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/build/Language/Haskell/Exts/Syntax.o )
    [ 5 of 18] Compiling Language.Haskell.Exts.ParseSyntax ( src/Language/Haskell/Exts/ParseSyntax.hs, .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/build/Language/Haskell/Exts/ParseSyntax.o )
    [ 6 of 18] Compiling Language.Haskell.Exts.Pretty ( src/Language/Haskell/Exts/Pretty.hs, .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/build/Language/Haskell/Exts/Pretty.o )
    [ 7 of 18] Compiling Language.Haskell.Exts.Fixity ( src/Language/Haskell/Exts/Fixity.hs, .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/build/Language/Haskell/Exts/Fixity.o )
    [ 8 of 18] Compiling Language.Haskell.Exts.Comments ( src/Language/Haskell/Exts/Comments.hs, .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/build/Language/Haskell/Exts/Comments.o )
    [ 9 of 18] Compiling Language.Haskell.Exts.ParseMonad ( src/Language/Haskell/Exts/ParseMonad.hs, .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/build/Language/Haskell/Exts/ParseMonad.o )
    [10 of 18] Compiling Language.Haskell.Exts.ParseUtils ( src/Language/Haskell/Exts/ParseUtils.hs, .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/build/Language/Haskell/Exts/ParseUtils.o )
    [11 of 18] Compiling Language.Haskell.Exts.InternalLexer ( src/Language/Haskell/Exts/InternalLexer.hs, .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/build/Language/Haskell/Exts/InternalLexer.o )
    [12 of 18] Compiling Language.Haskell.Exts.Lexer ( src/Language/Haskell/Exts/Lexer.hs, .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/build/Language/Haskell/Exts/Lexer.o )
    [13 of 18] Compiling Language.Haskell.Exts.InternalParser ( .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/build/Language/Haskell/Exts/InternalParser.hs, .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/build/Language/Haskell/Exts/InternalParser.o )
    [14 of 18] Compiling Language.Haskell.Exts.Parser ( src/Language/Haskell/Exts/Parser.hs, .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/build/Language/Haskell/Exts/Parser.o )
    [15 of 18] Compiling Language.Haskell.Exts.ExactPrint ( src/Language/Haskell/Exts/ExactPrint.hs, .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/build/Language/Haskell/Exts/ExactPrint.o )
    [16 of 18] Compiling Language.Haskell.Exts.Build ( src/Language/Haskell/Exts/Build.hs, .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/build/Language/Haskell/Exts/Build.o )
    [17 of 18] Compiling Language.Preprocessor.Unlit ( src/Language/Preprocessor/Unlit.hs, .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/build/Language/Preprocessor/Unlit.o )
    
    src/Language/Preprocessor/Unlit.hs:17:25: warning: [-Wname-shadowing]
        This binding for ‘x’ shadows the existing binding
          bound at src/Language/Preprocessor/Unlit.hs:14:17
       |
    17 |          allProg (('\\':x):xs) |  "end{code}"`isPrefixOf`x = Blank : classify xs
       |                         ^
    
    src/Language/Preprocessor/Unlit.hs:17:28: warning: [-Wname-shadowing]
        This binding for ‘xs’ shadows the existing binding
          bound at src/Language/Preprocessor/Unlit.hs:14:20
       |
    17 |          allProg (('\\':x):xs) |  "end{code}"`isPrefixOf`x = Blank : classify xs
       |                            ^^
    
    src/Language/Preprocessor/Unlit.hs:18:19: warning: [-Wname-shadowing]
        This binding for ‘x’ shadows the existing binding
          bound at src/Language/Preprocessor/Unlit.hs:14:17
       |
    18 |          allProg (x:xs) = Program x:allProg xs
       |                   ^
    
    src/Language/Preprocessor/Unlit.hs:18:21: warning: [-Wname-shadowing]
        This binding for ‘xs’ shadows the existing binding
          bound at src/Language/Preprocessor/Unlit.hs:14:20
       |
    18 |          allProg (x:xs) = Program x:allProg xs
       |                     ^^
    
    src/Language/Preprocessor/Unlit.hs:27:11: warning: [-Wunused-matches]
        Defined but not used: ‘x’
       |
    27 | classify (x:xs)                 = Comment:classify xs
       |           ^
    
    src/Language/Preprocessor/Unlit.hs:46:17: warning: [-Wunused-matches]
        Defined but not used: ‘y’
       |
    46 | adjacent file n y@(Program _) (x@Comment      :xs) = error (message file n "program" "comment")
       |                 ^
    
    src/Language/Preprocessor/Unlit.hs:46:32: warning: [-Wunused-matches]
        Defined but not used: ‘x’
       |
    46 | adjacent file n y@(Program _) (x@Comment      :xs) = error (message file n "program" "comment")
       |                                ^
    
    src/Language/Preprocessor/Unlit.hs:46:48: warning: [-Wunused-matches]
        Defined but not used: ‘xs’
       |
    46 | adjacent file n y@(Program _) (x@Comment      :xs) = error (message file n "program" "comment")
       |                                                ^^
    
    src/Language/Preprocessor/Unlit.hs:47:10: warning: [-Wunused-matches]
        Defined but not used: ‘file’
       |
    47 | adjacent file n y@(Program _) (x@(Include i f):xs) = x: adjacent f    i     y xs
       |          ^^^^
    
    src/Language/Preprocessor/Unlit.hs:47:15: warning: [-Wunused-matches]
        Defined but not used: ‘n’
       |
    47 | adjacent file n y@(Program _) (x@(Include i f):xs) = x: adjacent f    i     y xs
       |               ^
    
    src/Language/Preprocessor/Unlit.hs:49:17: warning: [-Wunused-matches]
        Defined but not used: ‘y’
       |
    49 | adjacent file n y@Comment     (x@(Program _)  :xs) = error (message file n "comment" "program")
       |                 ^
    
    src/Language/Preprocessor/Unlit.hs:49:32: warning: [-Wunused-matches]
        Defined but not used: ‘x’
       |
    49 | adjacent file n y@Comment     (x@(Program _)  :xs) = error (message file n "comment" "program")
       |                                ^
    
    src/Language/Preprocessor/Unlit.hs:49:48: warning: [-Wunused-matches]
        Defined but not used: ‘xs’
       |
    49 | adjacent file n y@Comment     (x@(Program _)  :xs) = error (message file n "comment" "program")
       |                                                ^^
    
    src/Language/Preprocessor/Unlit.hs:50:10: warning: [-Wunused-matches]
        Defined but not used: ‘file’
       |
    50 | adjacent file n y@Comment     (x@(Include i f):xs) = x: adjacent f    i     y xs
       |          ^^^^
    
    src/Language/Preprocessor/Unlit.hs:50:15: warning: [-Wunused-matches]
        Defined but not used: ‘n’
       |
    50 | adjacent file n y@Comment     (x@(Include i f):xs) = x: adjacent f    i     y xs
       |               ^
    
    src/Language/Preprocessor/Unlit.hs:52:10: warning: [-Wunused-matches]
        Defined but not used: ‘file’
       |
    52 | adjacent file n y@Blank       (x@(Include i f):xs) = x: adjacent f    i     y xs
       |          ^^^^
    
    src/Language/Preprocessor/Unlit.hs:52:15: warning: [-Wunused-matches]
        Defined but not used: ‘n’
       |
    52 | adjacent file n y@Blank       (x@(Include i f):xs) = x: adjacent f    i     y xs
       |               ^
    
    src/Language/Preprocessor/Unlit.hs:54:34: warning: [-Wunused-matches]
        Defined but not used: ‘next’
       |
    54 | adjacent file n _             (x@next         :xs) = x: adjacent file (n+1) x xs
       |                                  ^^^^
    
    src/Language/Preprocessor/Unlit.hs:55:10: warning: [-Wunused-matches]
        Defined but not used: ‘file’
       |
    55 | adjacent file n _             []                   = []
       |          ^^^^
    
    src/Language/Preprocessor/Unlit.hs:55:15: warning: [-Wunused-matches]
        Defined but not used: ‘n’
       |
    55 | adjacent file n _             []                   = []
       |               ^
    
    src/Language/Preprocessor/Unlit.hs:69:22: warning: [-Wname-shadowing]
        This binding for ‘s’ shadows the existing binding
          bound at src/Language/Preprocessor/Unlit.hs:66:9
       |
    69 |   lines' ('\^M':'\n':s) acc = acc [] : lines' s id      -- DOS
       |                      ^
    
    src/Language/Preprocessor/Unlit.hs:70:17: warning: [-Wname-shadowing]
        This binding for ‘s’ shadows the existing binding
          bound at src/Language/Preprocessor/Unlit.hs:66:9
       |
    70 |   lines' ('\^M':s)      acc = acc [] : lines' s id      -- MacOS
       |                 ^
    
    src/Language/Preprocessor/Unlit.hs:71:16: warning: [-Wname-shadowing]
        This binding for ‘s’ shadows the existing binding
          bound at src/Language/Preprocessor/Unlit.hs:66:9
       |
    71 |   lines' ('\n':s)       acc = acc [] : lines' s id      -- Unix
       |                ^
    
    src/Language/Preprocessor/Unlit.hs:72:13: warning: [-Wname-shadowing]
        This binding for ‘s’ shadows the existing binding
          bound at src/Language/Preprocessor/Unlit.hs:66:9
       |
    72 |   lines' (c:s)          acc = lines' s (acc . (c:))
       |             ^
    [18 of 18] Compiling Language.Haskell.Exts ( src/Language/Haskell/Exts.hs, .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/build/Language/Haskell/Exts.o )
    ignoring (possibly broken) abi-depends field for packages
    Preprocessing library for haskell-src-exts-1.20.3..
    Running Haddock on library for haskell-src-exts-1.20.3..
    
    src/Language/Preprocessor/Unlit.hs:17:25: warning: [-Wname-shadowing]
        This binding for ‘x’ shadows the existing binding
          bound at src/Language/Preprocessor/Unlit.hs:14:17
       |
    17 |          allProg (('\\':x):xs) |  "end{code}"`isPrefixOf`x = Blank : classify xs
       |                         ^
    
    src/Language/Preprocessor/Unlit.hs:17:28: warning: [-Wname-shadowing]
        This binding for ‘xs’ shadows the existing binding
          bound at src/Language/Preprocessor/Unlit.hs:14:20
       |
    17 |          allProg (('\\':x):xs) |  "end{code}"`isPrefixOf`x = Blank : classify xs
       |                            ^^
    
    src/Language/Preprocessor/Unlit.hs:18:19: warning: [-Wname-shadowing]
        This binding for ‘x’ shadows the existing binding
          bound at src/Language/Preprocessor/Unlit.hs:14:17
       |
    18 |          allProg (x:xs) = Program x:allProg xs
       |                   ^
    
    src/Language/Preprocessor/Unlit.hs:18:21: warning: [-Wname-shadowing]
        This binding for ‘xs’ shadows the existing binding
          bound at src/Language/Preprocessor/Unlit.hs:14:20
       |
    18 |          allProg (x:xs) = Program x:allProg xs
       |                     ^^
    
    src/Language/Preprocessor/Unlit.hs:27:11: warning: [-Wunused-matches]
        Defined but not used: ‘x’
       |
    27 | classify (x:xs)                 = Comment:classify xs
       |           ^
    
    src/Language/Preprocessor/Unlit.hs:46:17: warning: [-Wunused-matches]
        Defined but not used: ‘y’
       |
    46 | adjacent file n y@(Program _) (x@Comment      :xs) = error (message file n "program" "comment")
       |                 ^
    
    src/Language/Preprocessor/Unlit.hs:46:32: warning: [-Wunused-matches]
        Defined but not used: ‘x’
       |
    46 | adjacent file n y@(Program _) (x@Comment      :xs) = error (message file n "program" "comment")
       |                                ^
    
    src/Language/Preprocessor/Unlit.hs:46:48: warning: [-Wunused-matches]
        Defined but not used: ‘xs’
       |
    46 | adjacent file n y@(Program _) (x@Comment      :xs) = error (message file n "program" "comment")
       |                                                ^^
    
    src/Language/Preprocessor/Unlit.hs:47:10: warning: [-Wunused-matches]
        Defined but not used: ‘file’
       |
    47 | adjacent file n y@(Program _) (x@(Include i f):xs) = x: adjacent f    i     y xs
       |          ^^^^
    
    src/Language/Preprocessor/Unlit.hs:47:15: warning: [-Wunused-matches]
        Defined but not used: ‘n’
       |
    47 | adjacent file n y@(Program _) (x@(Include i f):xs) = x: adjacent f    i     y xs
       |               ^
    
    src/Language/Preprocessor/Unlit.hs:49:17: warning: [-Wunused-matches]
        Defined but not used: ‘y’
       |
    49 | adjacent file n y@Comment     (x@(Program _)  :xs) = error (message file n "comment" "program")
       |                 ^
    
    src/Language/Preprocessor/Unlit.hs:49:32: warning: [-Wunused-matches]
        Defined but not used: ‘x’
       |
    49 | adjacent file n y@Comment     (x@(Program _)  :xs) = error (message file n "comment" "program")
       |                                ^
    
    src/Language/Preprocessor/Unlit.hs:49:48: warning: [-Wunused-matches]
        Defined but not used: ‘xs’
       |
    49 | adjacent file n y@Comment     (x@(Program _)  :xs) = error (message file n "comment" "program")
       |                                                ^^
    
    src/Language/Preprocessor/Unlit.hs:50:10: warning: [-Wunused-matches]
        Defined but not used: ‘file’
       |
    50 | adjacent file n y@Comment     (x@(Include i f):xs) = x: adjacent f    i     y xs
       |          ^^^^
    
    src/Language/Preprocessor/Unlit.hs:50:15: warning: [-Wunused-matches]
        Defined but not used: ‘n’
       |
    50 | adjacent file n y@Comment     (x@(Include i f):xs) = x: adjacent f    i     y xs
       |               ^
    
    src/Language/Preprocessor/Unlit.hs:52:10: warning: [-Wunused-matches]
        Defined but not used: ‘file’
       |
    52 | adjacent file n y@Blank       (x@(Include i f):xs) = x: adjacent f    i     y xs
       |          ^^^^
    
    src/Language/Preprocessor/Unlit.hs:52:15: warning: [-Wunused-matches]
        Defined but not used: ‘n’
       |
    52 | adjacent file n y@Blank       (x@(Include i f):xs) = x: adjacent f    i     y xs
       |               ^
    
    src/Language/Preprocessor/Unlit.hs:54:34: warning: [-Wunused-matches]
        Defined but not used: ‘next’
       |
    54 | adjacent file n _             (x@next         :xs) = x: adjacent file (n+1) x xs
       |                                  ^^^^
    
    src/Language/Preprocessor/Unlit.hs:55:10: warning: [-Wunused-matches]
        Defined but not used: ‘file’
       |
    55 | adjacent file n _             []                   = []
       |          ^^^^
    
    src/Language/Preprocessor/Unlit.hs:55:15: warning: [-Wunused-matches]
        Defined but not used: ‘n’
       |
    55 | adjacent file n _             []                   = []
       |               ^
    
    src/Language/Preprocessor/Unlit.hs:69:22: warning: [-Wname-shadowing]
        This binding for ‘s’ shadows the existing binding
          bound at src/Language/Preprocessor/Unlit.hs:66:9
       |
    69 |   lines' ('\^M':'\n':s) acc = acc [] : lines' s id      -- DOS
       |                      ^
    
    src/Language/Preprocessor/Unlit.hs:70:17: warning: [-Wname-shadowing]
        This binding for ‘s’ shadows the existing binding
          bound at src/Language/Preprocessor/Unlit.hs:66:9
       |
    70 |   lines' ('\^M':s)      acc = acc [] : lines' s id      -- MacOS
       |                 ^
    
    src/Language/Preprocessor/Unlit.hs:71:16: warning: [-Wname-shadowing]
        This binding for ‘s’ shadows the existing binding
          bound at src/Language/Preprocessor/Unlit.hs:66:9
       |
    71 |   lines' ('\n':s)       acc = acc [] : lines' s id      -- Unix
       |                ^
    
    src/Language/Preprocessor/Unlit.hs:72:13: warning: [-Wname-shadowing]
        This binding for ‘s’ shadows the existing binding
          bound at src/Language/Preprocessor/Unlit.hs:66:9
       |
    72 |   lines' (c:s)          acc = lines' s (acc . (c:))
       |             ^

```

**Comment by qrilka** [comment]
It looks like the problem is in different settings in Stack - it hyperlinks sources by default, I get the same result with new cabal commands using the latest at this moment comit e0f2aa86d68b993f0a81633af467dc550b3e7270 -
```
$ cabal new-haddock --haddock-hyperlink-source
Build profile: -w ghc-8.4.4 -O1
In order, the following will be built (use -v for more details):
 - haskell-src-exts-1.21.0 (lib) (first run)
Preprocessing library for haskell-src-exts-1.21.0..
Running Haddock on library for haskell-src-exts-1.21.0..
Warning: --source-* options are ignored when --hyperlinked-source is enabled.
Haddock coverage:
  65% ( 13 / 20) in 'Language.Haskell.Exts.Extension'
  Missing documentation for:
    Language (src/Language/Haskell/Exts/Extension.hs:86)
    knownLanguages (src/Language/Haskell/Exts/Extension.hs:105)
    classifyLanguage (src/Language/Haskell/Exts/Extension.hs:108)
    prettyLanguage (src/Language/Haskell/Exts/Extension.hs:114)
    KnownExtension (src/Language/Haskell/Exts/Extension.hs:135)
    ghcDefault (src/Language/Haskell/Exts/Extension.hs:642)
    toExtensionList (src/Language/Haskell/Exts/Extension.hs:721)
  91% ( 21 / 23) in 'Language.Haskell.Exts.SrcLoc'
  Missing documentation for:
    noLoc (src/Language/Haskell/Exts/SrcLoc.hs:34)
    spanSize (src/Language/Haskell/Exts/SrcLoc.hs:82)
  65% ( 99 /152) in 'Language.Haskell.Exts.Syntax'
  Missing documentation for:
    PatternSynDirection (src/Language/Haskell/Exts/Syntax.hs:344)
    ResultSig (src/Language/Haskell/Exts/Syntax.hs:387)
    Unpackedness (src/Language/Haskell/Exts/Syntax.hs:603)
    Role (src/Language/Haskell/Exts/Syntax.hs:371)
    MaybePromotedName (src/Language/Haskell/Exts/Syntax.hs:653)
    prelude_mod (src/Language/Haskell/Exts/Syntax.hs:1010)
    main_mod (src/Language/Haskell/Exts/Syntax.hs:1010)
    main_name (src/Language/Haskell/Exts/Syntax.hs:1014)
    unit_con_name (src/Language/Haskell/Exts/Syntax.hs:1017)
    tuple_con_name (src/Language/Haskell/Exts/Syntax.hs:1020)
    list_con_name (src/Language/Haskell/Exts/Syntax.hs:1023)
    list_cons_name (src/Language/Haskell/Exts/Syntax.hs:1026)
    unboxed_singleton_con_name (src/Language/Haskell/Exts/Syntax.hs:1029)
    unit_con (src/Language/Haskell/Exts/Syntax.hs:1032)
    tuple_con (src/Language/Haskell/Exts/Syntax.hs:1035)
    unboxed_singleton_con (src/Language/Haskell/Exts/Syntax.hs:1038)
    as_name (src/Language/Haskell/Exts/Syntax.hs:1041)
    qualified_name (src/Language/Haskell/Exts/Syntax.hs:1041)
    hiding_name (src/Language/Haskell/Exts/Syntax.hs:1041)
    minus_name (src/Language/Haskell/Exts/Syntax.hs:1041)
    bang_name (src/Language/Haskell/Exts/Syntax.hs:1041)
    dot_name (src/Language/Haskell/Exts/Syntax.hs:1041)
    star_name (src/Language/Haskell/Exts/Syntax.hs:1041)
    export_name (src/Language/Haskell/Exts/Syntax.hs:1053)
    safe_name (src/Language/Haskell/Exts/Syntax.hs:1053)
    unsafe_name (src/Language/Haskell/Exts/Syntax.hs:1053)
    interruptible_name (src/Language/Haskell/Exts/Syntax.hs:1053)
    threadsafe_name (src/Language/Haskell/Exts/Syntax.hs:1053)
    stdcall_name (src/Language/Haskell/Exts/Syntax.hs:1053)
    ccall_name (src/Language/Haskell/Exts/Syntax.hs:1053)
    cplusplus_name (src/Language/Haskell/Exts/Syntax.hs:1053)
    dotnet_name (src/Language/Haskell/Exts/Syntax.hs:1053)
    jvm_name (src/Language/Haskell/Exts/Syntax.hs:1053)
    js_name (src/Language/Haskell/Exts/Syntax.hs:1053)
    javascript_name (src/Language/Haskell/Exts/Syntax.hs:1053)
    capi_name (src/Language/Haskell/Exts/Syntax.hs:1053)
    forall_name (src/Language/Haskell/Exts/Syntax.hs:1053)
    family_name (src/Language/Haskell/Exts/Syntax.hs:1053)
    role_name (src/Language/Haskell/Exts/Syntax.hs:1053)
    hole_name (src/Language/Haskell/Exts/Syntax.hs:1050)
    stock_name (src/Language/Haskell/Exts/Syntax.hs:1053)
    anyclass_name (src/Language/Haskell/Exts/Syntax.hs:1053)
    via_name (src/Language/Haskell/Exts/Syntax.hs:1053)
    unit_tycon_name (src/Language/Haskell/Exts/Syntax.hs:1077)
    fun_tycon_name (src/Language/Haskell/Exts/Syntax.hs:1077)
    list_tycon_name (src/Language/Haskell/Exts/Syntax.hs:1077)
    tuple_tycon_name (src/Language/Haskell/Exts/Syntax.hs:1083)
    unboxed_singleton_tycon_name (src/Language/Haskell/Exts/Syntax.hs:1077)
    unit_tycon (src/Language/Haskell/Exts/Syntax.hs:1086)
    fun_tycon (src/Language/Haskell/Exts/Syntax.hs:1086)
    list_tycon (src/Language/Haskell/Exts/Syntax.hs:1086)
    tuple_tycon (src/Language/Haskell/Exts/Syntax.hs:1092)
    unboxed_singleton_tycon (src/Language/Haskell/Exts/Syntax.hs:1086)
  94% ( 17 / 18) in 'Language.Haskell.Exts.Pretty'
  Missing documentation for:
    Indent (src/Language/Haskell/Exts/Pretty.hs:58)
  75% (  9 / 12) in 'Language.Haskell.Exts.Fixity'
  Missing documentation for:
    infix_ (src/Language/Haskell/Exts/Fixity.hs:184)
    infixl_ (src/Language/Haskell/Exts/Fixity.hs:184)
    infixr_ (src/Language/Haskell/Exts/Fixity.hs:184)
 100% (  4 /  4) in 'Language.Haskell.Exts.Comments'
  67% (  4 /  6) in 'Language.Haskell.Exts.Lexer'
  Missing documentation for:
    Token (src/Language/Haskell/Exts/InternalLexer.hs:37)
    showToken (src/Language/Haskell/Exts/InternalLexer.hs:1291)
  93% ( 42 / 45) in 'Language.Haskell.Exts.Parser'
  Missing documentation for:
    unListOf (src/Language/Haskell/Exts/Parser.hs:249)
    PragmasAndModuleHead (src/Language/Haskell/Exts/Parser.hs:192)
    ModuleHeadAndImports (src/Language/Haskell/Exts/Parser.hs:212)
  67% (  2 /  3) in 'Language.Haskell.Exts.ExactPrint'
  Missing documentation for:
    ExactP (src/Language/Haskell/Exts/ExactPrint.hs:220)
 100% ( 49 / 49) in 'Language.Haskell.Exts.Build'

src/Language/Preprocessor/Unlit.hs:17:25: warning: [-Wname-shadowing]
    This binding for ‘x’ shadows the existing binding
      bound at src/Language/Preprocessor/Unlit.hs:14:17
   |
17 |          allProg (('\\':x):xs) |  "end{code}"`isPrefixOf`x = Blank : classify xs
   |                         ^

src/Language/Preprocessor/Unlit.hs:17:28: warning: [-Wname-shadowing]
    This binding for ‘xs’ shadows the existing binding
      bound at src/Language/Preprocessor/Unlit.hs:14:20
   |
17 |          allProg (('\\':x):xs) |  "end{code}"`isPrefixOf`x = Blank : classify xs
   |                            ^^

src/Language/Preprocessor/Unlit.hs:18:19: warning: [-Wname-shadowing]
    This binding for ‘x’ shadows the existing binding
      bound at src/Language/Preprocessor/Unlit.hs:14:17
   |
18 |          allProg (x:xs) = Program x:allProg xs
   |                   ^

src/Language/Preprocessor/Unlit.hs:18:21: warning: [-Wname-shadowing]
    This binding for ‘xs’ shadows the existing binding
      bound at src/Language/Preprocessor/Unlit.hs:14:20
   |
18 |          allProg (x:xs) = Program x:allProg xs
   |                     ^^

src/Language/Preprocessor/Unlit.hs:27:11: warning: [-Wunused-matches]
    Defined but not used: ‘x’
   |
27 | classify (x:xs)                 = Comment:classify xs
   |           ^

src/Language/Preprocessor/Unlit.hs:46:17: warning: [-Wunused-matches]
    Defined but not used: ‘y’
   |
46 | adjacent file n y@(Program _) (x@Comment      :xs) = error (message file n "program" "comment")
   |                 ^

src/Language/Preprocessor/Unlit.hs:46:32: warning: [-Wunused-matches]
    Defined but not used: ‘x’
   |
46 | adjacent file n y@(Program _) (x@Comment      :xs) = error (message file n "program" "comment")
   |                                ^

src/Language/Preprocessor/Unlit.hs:46:48: warning: [-Wunused-matches]
    Defined but not used: ‘xs’
   |
46 | adjacent file n y@(Program _) (x@Comment      :xs) = error (message file n "program" "comment")
   |                                                ^^

src/Language/Preprocessor/Unlit.hs:47:10: warning: [-Wunused-matches]
    Defined but not used: ‘file’
   |
47 | adjacent file n y@(Program _) (x@(Include i f):xs) = x: adjacent f    i     y xs
   |          ^^^^

src/Language/Preprocessor/Unlit.hs:47:15: warning: [-Wunused-matches]
    Defined but not used: ‘n’
   |
47 | adjacent file n y@(Program _) (x@(Include i f):xs) = x: adjacent f    i     y xs
   |               ^

src/Language/Preprocessor/Unlit.hs:49:17: warning: [-Wunused-matches]
    Defined but not used: ‘y’
   |
49 | adjacent file n y@Comment     (x@(Program _)  :xs) = error (message file n "comment" "program")
   |                 ^

src/Language/Preprocessor/Unlit.hs:49:32: warning: [-Wunused-matches]
    Defined but not used: ‘x’
   |
49 | adjacent file n y@Comment     (x@(Program _)  :xs) = error (message file n "comment" "program")
   |                                ^

src/Language/Preprocessor/Unlit.hs:49:48: warning: [-Wunused-matches]
    Defined but not used: ‘xs’
   |
49 | adjacent file n y@Comment     (x@(Program _)  :xs) = error (message file n "comment" "program")
   |                                                ^^

src/Language/Preprocessor/Unlit.hs:50:10: warning: [-Wunused-matches]
    Defined but not used: ‘file’
   |
50 | adjacent file n y@Comment     (x@(Include i f):xs) = x: adjacent f    i     y xs
   |          ^^^^

src/Language/Preprocessor/Unlit.hs:50:15: warning: [-Wunused-matches]
    Defined but not used: ‘n’
   |
50 | adjacent file n y@Comment     (x@(Include i f):xs) = x: adjacent f    i     y xs
   |               ^

src/Language/Preprocessor/Unlit.hs:52:10: warning: [-Wunused-matches]
    Defined but not used: ‘file’
   |
52 | adjacent file n y@Blank       (x@(Include i f):xs) = x: adjacent f    i     y xs
   |          ^^^^

src/Language/Preprocessor/Unlit.hs:52:15: warning: [-Wunused-matches]
    Defined but not used: ‘n’
   |
52 | adjacent file n y@Blank       (x@(Include i f):xs) = x: adjacent f    i     y xs
   |               ^

src/Language/Preprocessor/Unlit.hs:54:34: warning: [-Wunused-matches]
    Defined but not used: ‘next’
   |
54 | adjacent file n _             (x@next         :xs) = x: adjacent file (n+1) x xs
   |                                  ^^^^

src/Language/Preprocessor/Unlit.hs:55:10: warning: [-Wunused-matches]
    Defined but not used: ‘file’
   |
55 | adjacent file n _             []                   = []
   |          ^^^^

src/Language/Preprocessor/Unlit.hs:55:15: warning: [-Wunused-matches]
    Defined but not used: ‘n’
   |
55 | adjacent file n _             []                   = []
   |               ^

src/Language/Preprocessor/Unlit.hs:69:22: warning: [-Wname-shadowing]
    This binding for ‘s’ shadows the existing binding
      bound at src/Language/Preprocessor/Unlit.hs:66:9
   |
69 |   lines' ('\^M':'\n':s) acc = acc [] : lines' s id      -- DOS
   |                      ^

src/Language/Preprocessor/Unlit.hs:70:17: warning: [-Wname-shadowing]
    This binding for ‘s’ shadows the existing binding
      bound at src/Language/Preprocessor/Unlit.hs:66:9
   |
70 |   lines' ('\^M':s)      acc = acc [] : lines' s id      -- MacOS
   |                 ^

src/Language/Preprocessor/Unlit.hs:71:16: warning: [-Wname-shadowing]
    This binding for ‘s’ shadows the existing binding
      bound at src/Language/Preprocessor/Unlit.hs:66:9
   |
71 |   lines' ('\n':s)       acc = acc [] : lines' s id      -- Unix
   |                ^

src/Language/Preprocessor/Unlit.hs:72:13: warning: [-Wname-shadowing]
    This binding for ‘s’ shadows the existing binding
      bound at src/Language/Preprocessor/Unlit.hs:66:9
   |
72 |   lines' (c:s)          acc = lines' s (acc . (c:))
   |             ^
  88% ( 23 / 26) in 'Language.Haskell.Exts'
  Missing documentation for:
    Token (src/Language/Haskell/Exts/InternalLexer.hs:37)
    parseFileWithComments (src/Language/Haskell/Exts.hs:74)
    parseFileContentsWithComments (src/Language/Haskell/Exts.hs:115)
Warning: Language.Haskell.Exts.SrcLoc: could not find link destinations for:
    pretty Doc prettyPrec parser P
Warning: Language.Haskell.Exts.Syntax: could not find link destinations for:
    exactP EP pretty Doc prettyPrec parser P
Warning: Language.Haskell.Exts.Pretty: could not find link destinations for:
    pretty Doc prettyPrec
Warning: Language.Haskell.Exts.Parser: could not find link destinations for:
    parser P
Warning: Language.Haskell.Exts.ExactPrint: could not find link destinations for:
    exactP EP
cabal: Failed to build documentation for haskell-src-exts-1.21.0. The build
process segfaulted (i.e. SIGSEGV).
```

**Comment by mpickering** [comment]
Seems like a bug in `haddock` to me?

**Comment by qrilka** [comment]
Looks like it was resolved in GHC 8.6.1

**Comment by damiencourousse** [comment]
yes, it's also fine for me with lts-13.0
